backend/wallet/services/offchain/utils.py

```diff
@@ -79,3 +85,46 @@ def jws_response(
         err=err,
     )
     return code, offchain.jws.serialize(resp, compliance_private_key().sign)
+
+
+def get_p2m_metadata_by_ref_id(reference_id) -> Metadata:
+    metadata_bytes = txnmetadata.payment_metadata(reference_id)
+    return diem_types.Metadata__TravelRuleMetadata.bcs_deserialize(metadata_bytes)
+
+
+# submit transaction on-chain using p2m metadata.
+# recipient signature is needed only if the amount >= 1000 (travel rule)
+# input example: ref-id: 1c5ee1eb-559b-48b3-898c-85b4d4713890,
+#               amount: 100000000,
+#               currency: XUS,
+#               recipient_signature: None,
+#               receiver_address: tdm1paue4j3fqr6nzl5xglqr337surful4nmrx6kn9fgmh5qz6
+def submit_p2m_txn(reference_id: str,
```

Comment:
Who is using this method? 
I think no offchain service should interact with the blockchain directly.
I see that you moved the actual submission to the transaction service (good call), but why did you leave it here?